### PR TITLE
First solution to week0002 problem

### PR DESCRIPTION
**Pull request by hoangdangduy**
Please review for me. Thanks!

### CODE REVIEW

File: week0002/hoangdangduy/java/Week0002.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+		for (int temp1 = 0; temp1 < M; temp1++) {
+			for (int temp2 = 0; temp2 < N; temp2++) {
+				System.out.print(arr[temp1][temp2] + "   ");
+			}
+			System.out.println();
+		}
+	}
+	
+	public static void main(String[] args) {
```
Comment (vietlq) [inline week0002/hoangdangduy/java/Week0002.java]:
```
$ javac Week0002.java
$ java Week0002
Error: Could not find or load main class Week0002
```

File: week0002/hoangdangduy/java/Week0002.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+		Week0002 week0002 = new Week0002();
+		week0002.Spiral(1, 2);
```
Comment (vietlq) [inline week0002/hoangdangduy/java/Week0002.java]: We agreed to print 1-16 :)

**Comment by hoangdangduy** [comment]
now, do I must remove package and re-commit?
thanks.

File: week0002/hoangdangduy/java/Week0002.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+			bottomLeftCorner++;
+			i--;
+			j++;
+
+			/*
+			 * run from below to above
+			 */
+			while (i > upperLeftConrner && numRun <= (M * N)) {
+				arr[i][j] = numRun++;
+				// System.out.println(i + " - " + j + " - " + arr[i][j]);
```
Comment (vietlq) [inline week0002/hoangdangduy/java/Week0002.java]: Remove unused code

File: week0002/hoangdangduy/java/Week0002.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+			upperRightConrner--;
+			j--;
+			i++;
+
+			/*
+			 * run from above to below
+			 */
+			while (i <= bottomRightCorner && numRun <= (M * N)) {
+				arr[i][j] = numRun++;
+				// System.out.println(i + " - " + j + " - " + arr[i][j]);
```
Comment (vietlq) [inline week0002/hoangdangduy/java/Week0002.java]: Remove unused code

File: week0002/hoangdangduy/java/Week0002.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+			return;
+		}
+
+		while (numRun <= (M * N)) {
+			/*
+			 * run from left to right
+			 */
+			while (j <= upperRightConrner && numRun <= (M * N)) {
+				arr[i][j] = numRun++;
+				// System.out.println(i + " - " + j + " - " + arr[i][j]);
```
Comment (vietlq) [inline week0002/hoangdangduy/java/Week0002.java]: Remove unused code

File: week0002/hoangdangduy/java/Week0002.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+				j++;
+			}
+			upperRightConrner--;
```
Comment (vietlq) [inline week0002/hoangdangduy/java/Week0002.java]: spelling: `corner`

File: week0002/hoangdangduy/java/Week0002.java [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+			/*
+			 * run from below to above
+			 */
+			while (i > upperLeftConrner && numRun <= (M * N)) {
+				arr[i][j] = numRun++;
+				// System.out.println(i + " - " + j + " - " + arr[i][j]);
+				i--;
+			}
+			i++;
+			upperLeftConrner++;
```
Comment (vietlq) [inline week0002/hoangdangduy/java/Week0002.java]: Spelling again.

**Pull request merged**